src/async_impl/client.rs

```diff
@@ -591,6 +594,19 @@ impl ClientBuilder {
 
     // HTTP options
 
+    /// Set an optional timeout for idle sockets being kept-alive.
+    ///
+    /// Pass `None` to disable timeout.
+    ///
+    /// Default is 90 seconds.
+    pub fn idle_timeout<D>(mut self, val: D) -> ClientBuilder
```

Comment:
A tiny thought: perhaps we should name this `pool_idle_timeout`, like hyper does.

Reply:
Good point, and thanks for taking the time to review this! In that case, I’d suggest renaming `max_idle_per_host` to `pool_max_idle_per_host` (which is the name `hyper` uses) to maintain consistency.

Reply:
That sounds like a good change too. If you want to make a PR with that change, we'd just need to `#[deprecate]` and `#[doc(hidden)]` the old name.

Reply:
Perfect :+1:! I’ll take care of changing this pull request and submitting another one to rename `max_idle_per_host` to `pool_max_idle_per_host`.